daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Decimal.scala

```diff
@@ -28,24 +27,45 @@ object Decimal {
 
   val min: Decimal = unlimitedBigDecimal("-9999999999999999999999999999.9999999999")
 
-  def check(x0: Decimal): Either[String, Decimal] = {
+  /** Checks that a `Decimal` falls between `min` and `max`, and
+    * round the number according to `scale`. Note that it does _not_
+    * fail if the number contains data beyond `scale`.
+    */
+  def checkWithinBoundsAndRound(x0: Decimal): Either[String, Decimal] = {
+    val err = Left(s"out-of-bounds Decimal $x0")
     if (x0 > max || x0 < min) {
-      Left(s"out-of-bounds Decimal $x0")
```

Comment:
I will have let the creation of the left inside the `if` as it is not use anywhere else